Adds new terms

Tiaraha-funny opened the pull request:
Hello team,

In Add-new-term branch handle the add screen for the user to select a category that they want to add in Malagasy and English term. Storage component is created to store the added category.  In addition, add new term container contains the actions and pass the redux reducers.

If you could, would you mind reviewing it and give some feedback if needed. I would appreciate all the changes.

Thanks

https://app.clickup.com/t/35bjay

lightme-fan commented on src/utils/storage.js:
```diff
@@ -0,0 +1,21 @@
+  }
+}
+
+// AsyncStorage.clear();
```
Remove this piece of code.

lightme-fan reviewed (commented):
It's good. It may be good if you don't use inline styles. You could use styled components if you want.

lightme-fan committed:
feat(Add Screen): Add New Term Screen
Add storage componen
Adds add screen in navigation
Adds add new term component

Refs: U4#35bjay

 #refactor(learnt Phrases): refactored this code that toogle the category name in the learning screen.

andrzej-onja commented on src/redux/actions/index.js:
```diff
@@ -1,11 +1,78 @@
 
+// Adding new term actions
+
+export function setUserPhrases(newTerm) {
```
change `newTerm` to `phrases`

andrzej-onja commented on src/screens/AddNewTerm.js:
```diff
@@ -0,0 +1,193 @@
+
+  function malagasyWordInput(e) {
+    setMalagasyTerm(e);
+  }
```
instead of the above use useState. EnglishTerm and MalagasyTerm dont have to be in redux state as no other components are using those values

Tiaraha-funny replied:
I have made the change and instead of creating a function I uses (e) => setState(e) which makes the code smaller.

Thanks.

Betsimisaraka reviewed (approved):
It's looking great for me.

andrzej-onja commented on package.json:
```diff
@@ -13,20 +13,29 @@
     "react-redux": "^7.2.4",
     "redux": "^4.1.0",
+    "redux-devtools-extension": "^2.13.9",
+    "redux-persist": "^6.0.0",
```
Remove package if unused

andrzej-onja commented on package.json:
```diff
@@ -13,20 +13,29 @@
     "react": "17.0.1",
     "react-native": "0.64.0",
     "react-native-dev-menu": "^4.0.2",
+    "react-native-drop-down-item": "^1.1.0",
```
Remove package if unused

andrzej-onja commented on package.json:
```diff
@@ -13,20 +13,29 @@
+    "@react-native-async-storage/async-storage": "^1.15.4",
+    "@codler/react-native-keyboard-aware-scroll-view": "^1.0.1",
+    "@react-native-community/async-storage": "^1.12.1",
+    "@react-native-community/picker": "^1.8.1",
```
Remove package if unused

Tiaraha-funny replied:
I didn't delete the "@codler/react-native-keyboard-aware-scroll-view": "^1.0.1" because I am using it with quiting keyboard.

andrzej-onja commented on package.json:
```diff
@@ -13,20 +13,29 @@
+    "@codler/react-native-keyboard-aware-scroll-view": "^1.0.1",
+    "@react-native-community/async-storage": "^1.12.1",
+    "@react-native-community/picker": "^1.8.1",
+    "@react-native-picker/picker": "^1.16.0",
```
Remove package if unused

andrzej-onja commented on src/redux/actions/index.js:
```diff
@@ -34,3 +44,75 @@ export function setLanguageName(language) {
+    dispatch(setLearntPhrases(dataToStore));
+    return Promise.resolve();
+  };
+}
```
Why this is here ?
`addLearntPhrase` is part of the other task, this branch should have changes only related to 'adding new phrases'

andrzej-onja commented on src/redux/actions/index.js:
```diff
@@ -34,3 +44,75 @@ export function setLanguageName(language) {
+    dispatch(setUserPhrases(storedPhrases));
+    if (!storedLearntPhrase) {
+      return Promise.resolve();
+    }
```
same here remove changes related to LearntPhrase

andrzej-onja commented on src/redux/reducers/index.js:
```diff
@@ -46,10 +49,40 @@ function nativeLanguage(state = '', action) {
+    default: 
+    return state;
+  }
+}
```
same here remove changes related to LearntPhrase

andrzej-onja commented on src/screens/AddNewTerm.js:
```diff
@@ -0,0 +1,184 @@
+
+            <View style={{marginTop: 45}}>
+              <NextButton
+                isDisabled={englishTerm && malagasyTerm ? false : true}
```
it can be just `isDisabled={!Boolean(englishTerm && malagasyTerm)}`

andrzej-onja commented on src/screens/Learning.js:
```diff
@@ -48,14 +52,15 @@ export default ({
+        learntPhrases?.every(phrase => phrase.id !== currentPhrase.id)
+      ) {
+        addLearntPhrase(item);
       } else {
```
keep `// TODO add to seen` as its not implemented yet

andrzej-onja commented on src/screens/Learning.js:
```diff
@@ -90,10 +96,15 @@ export default ({
+    cat.phrasesIds.includes(currentPhrase?.id),
+  );
+
+  const learntCatName = learntCategory?.name.en;
```
not part of this task

andrzej-onja commented on src/screens/Learning.js:
```diff
@@ -154,7 +171,7 @@ export default ({
               </View>
               <List
-                lang={LANGUAGE_NAMES.MG}
+                lang={LANGUAGE_NAMES.EN}
```
not part of this task ?

andrzej-onja commented on src/utils/storage.js:
```diff
@@ -0,0 +1,20 @@
+  } catch (e) {
+    console.log(e);
+  }
+}
```
You have to pick one implementation for interacting with AsyncStorage
`src/utils/storage.js` or `src/utils/localStorage.js`

andrzej-onja commented on src/redux/actions/index.js:
```diff
@@ -34,3 +44,75 @@ export function setLanguageName(language) {
+    type: SELECTE_VALUE,
+    payload: category,
+  };
+}
```
We dont need to store this in redux, its needed only for AddNewPhrase screen

Commit:
fix bug

andrzej-onja commented on src/redux/actions/index.js:
```diff
@@ -47,22 +55,52 @@ export function setLearntPhrases(learntPhrases) {
+    dispatch(setUserPhrases(storedPhrases));
     if (!storedLearntPhrase) {
       return Promise.resolve();
     }
```
it should be as below to not return before all store items are checked
 
  ```const storedPhrases = await getStoreItem(USER_PHRASES_KEY);
    const storedLearntPhrase = await getStoreItem(LEARNT_PHRASES_KEY);
    if (storedPhrases) {
       dispatch(setUserPhrases(storedPhrases));
    }
   
    if (storedLearntPhrase) {
          dispatch(setLearntPhrases(storedLearntPhrase));
    }

    return Promise.resolve();```

andrzej-onja commented on src/screens/Home.js:
```diff
@@ -49,14 +51,27 @@ export default ({
+      return
+    } else {
+    navigation.navigate('Learn');
+    }
```
lets invert condition so its more readable
```
 if(learntPhrases?.length>  0) {
      navigation.navigate('Learn');
    }
```

constantingoeldel commented:
You need to resolve the merge conflict and either address or resolve Andrzejs comment from 8 days ago, then you can merge

Commit:
deleted unused packages

Commit:
deleted a line of code

Commit:
deleted a line of code
